ui/tests/unit/components/filter/Permission.spec.js

```diff
@@ -0,0 +1,100 @@
+import permission from '../../../../src/components/filter/permission';
+import { actions, authorizer } from '../../../../src/authorizer';
+
+const ownerPermissionToTest = [
+  actions.device.add,
+  actions.device.accept,
+  actions.device.reject,
+  actions.device.remove,
+  actions.device.connect,
+  actions.device.rename,
+  actions.device.details,
+  actions.session.play,
+  actions.session.close,
+  actions.session.details,
+  actions.session.removeRecord,
+  actions.firewall.create,
+  actions.firewall.edit,
+  actions.firewall.remove,
+  actions.publicKey.create,
+  actions.publicKey.edit,
+  actions.publicKey.remove,
+  actions.namespace.create,
+  actions.namespace.rename,
+  actions.namespace.addMember,
+  actions.namespace.removeMember,
+  actions.namespace.enableSessionRecord,
+  actions.namespace.remove,
+  actions.billing.subscribe,
+  actions.billing.unsubscribe,
+  actions.notification.view,
+];
+const administratorPermissionToTest = [
+  actions.device.add,
+  actions.device.accept,
+  actions.device.reject,
+  actions.device.remove,
+  actions.device.connect,
+  actions.device.rename,
+  actions.device.details,
+  actions.session.play,
+  actions.session.close,
+  actions.session.details,
+  actions.session.removeRecord,
+  actions.firewall.create,
+  actions.firewall.edit,
+  actions.firewall.remove,
+  actions.publicKey.create,
+  actions.publicKey.edit,
+  actions.publicKey.remove,
+  actions.namespace.create,
+  actions.namespace.rename,
+  actions.namespace.addMember,
+  actions.namespace.removeMember,
+  actions.namespace.enableSessionRecord,
+  actions.namespace.remove,
+  actions.billing.subscribe,
+  actions.billing.unsubscribe,
+  actions.notification.view,
+];
+const operatorPermissionToTest = [
+  actions.device.add,
+  actions.device.accept,
+  actions.device.reject,
+  actions.device.connect,
+  actions.device.rename,
+  actions.device.details,
+  actions.session.details,
+  actions.namespace.create,
+];
+const observerPermissionsToTest = [
+  actions.device.connect,
+  actions.device.details,
+  actions.session.details,
+  actions.namespace.create,
+];
+
+const casesPermissions = [
+  [true, 'observer', authorizer.permissions.observer, observerPermissionsToTest],
+  [true, 'administrator', authorizer.permissions.administrator, administratorPermissionToTest],
+  [true, 'operator', authorizer.permissions.operator, operatorPermissionToTest],
+  [true, 'owner', authorizer.permissions.owner, ownerPermissionToTest],
+];
+
+const casesInvalid = [
+  [false, -1, -1],
+  [false, 'valid', -1],
+  [false, 'valid', 26],
+  [false, -1, 'valid'],
+  [false, 5, 'valid'],
+];
+
+describe('Permission', () => {
```

Comment:
Here you test a custom permission list; I fail to see why it uses the `authorizer.permissions` here. For me, it seems it could just validate three cases:
1. valid
2. invalid
3. non-existent

If those works, considering `authorizer.permissions` has the proper format, it'll work.

Reply:
I've thought that if the local permission, "mocked", is not equal permission used by the UI, the tests should break.

Reply:
In my understanding, the reason to use a mock is to not depend on external values so I think it could exercise the needed cases only. @gustavosbarreto?